Bug #22 - Make "Add" button put color in palette on 1st click:

marcioreisjr opened the pull request:
Hello Jacob and an folks reviewing this code,

This change is more a "button behavioral" change than technical. The rationale is that the "Add" button should add colors to the bar only after the colors are already defined, and the "Palette" button should define the primary or secondary color based on the selected color.

Currently, the "Add" button triggers some code execution even if there is no previously selected color, exhibiting the undesired behavior listed on this bug.

One solution is to remove this button as suggested in my previous comment<https://github.com/jb-tlm/color-charter/issues/22#issuecomment-1639080919> because case #13 added the ability to click on any color to add it to the Color Bar.

The proposed solution is to allow the "Add" button to work only after selecting a color with the "Palette" button.

Please, let me know if you prefer to remove the button or otherwise by accepting this PR will *conditionally* enable the button.

marcioreisjr committed:
Bug #22 - Make "Add" button put color in palette on 1st click:
## index.html:
- Set the button type as "button" to prevent it from generating the "submit" event.

## toolbar.js:
- Fixed the conditional that prevented the button from pushing an "empty" color to the Color Bar.

jb-tlm reviewed (commented):
Marcio, I think that the Add button should add the key color to the palette above anytime it's clicked. Currently if you hit the color button, select a color, hit the palette button, and then click Add, you will get your key color added to the palette above. However, if you then click the color button and change to a different color, clicking the Add button does nothing. This will seem like a bug to users. Unless they first click the palette button and have the palette generated, this new color won't be added to the palette above.
This logic shouldn't be set that way. The add button should read if there is a key color currently selected, and if it isn't already the most recent color in the palette, adds that key color to the palette. 
The palette button should be a separate tool that students can use to generate palettes based on relationships and then individually click those to add them to their palette. It seems to me that the Add button should be pretty straightforward and always add the color unless it's already added. The requirement to first generate the palette before this button works can seem confusing and like a bug in some situations.

marcioreisjr committed:
Addedssed comment <https://github.com/jb-tlm/color-charter/pull/28#pullrequestreview-1538007618>:
## main.js:
- Enabled the palette button to work after the first color insertion.

## toolbar.js:
- Changed the source of color from the <span> with the color code to the value on the color selector upon populating a new color to the bar.
